websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now());
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			presenceData.smallImageText = "sonicretro.org";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (search === "?s=")
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.title
+							.split("You searched for ")[1]
+							.split(" - Sonic Retro")[0]
+					}"`;
+				}
+			} else if (
+				(pathname === ("/" || "/#") && !search) ||
+				(pathname.includes("/page/") && !search)
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${document.title
+					.split("-")[0]
+					.toLowerCase()}`;
+			} else if (
+				document.title.match(
+					/January|February|March|April|May|June|July|August|September|October|November|December/g
+				)
+			) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking for posts in/on ${
+					document.title.split("-")[0]
+				}`;
+			} else if (pathname.includes("/wp-content/")) {
+				// viewing uploaded content; unnecesary to show in the presence
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (
+				location.href.match(/\/([0-9]+(\/[0-9]+)+)\/+((\w|-|\d)*)\//g)
+			) {
```

Comment:
never use String.match for if cases
```suggestion
			} else if (
				/\/([0-9]+(\/[0-9]+)+)\/+((\w|-|\d)*)\//g.test(location.href)
			) {
```